provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
On 21 June 2016 at 17:56, Fabian Reinartz <notifications@github.com> wrote:
>
> Meh, think we discussed these things in the past. Golint has way less
> confidence than govet and commenting by saying "NewSilencer returns a new silencer"
> seems slightly useless.

We might have discussed these things, but apparently, we came to
different conclusions:

- Yes, doc comments sometimes appear useless.
- Still forcing oneself to write them down has the advantage of
thinking twice if there is a way to create a useful doc comment after
all.
- Furthermore, it silences a golint warning. Golint warnings might be
false positives, but easily avoided warnings should still be avoided,
so that we don't create useless noise and are able to spot meaningful
golint warnings.

All provider sub packages are, btw, completely free of golint
warnings. Those two are the only ones.






-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales
with Company No. 6343600 | Local Branch Office | AG Charlottenburg  |
HRB 110657B